common/containers/Tabs/SendTransaction/index.jsx

```diff
@@ -238,8 +267,63 @@ export class SendTransaction extends React.Component {
     return { to, data, value, unit, gasLimit, readOnly };
   }
 
+  isValid() {
+    const { to, value } = this.state;
+    return (
+      isValidETHAddress(to) &&
+      value &&
+      Number(value) > 0 &&
+      !isNaN(Number(value)) &&
+      isFinite(Number(value))
+    );
+  }
+
+  // FIXME MOVE ME
+  getTransactionFromState() {
+    // FIXME add gas price
+    if (this.state.unit === 'ether') {
+      return {
+        to: this.state.to,
+        from: this.props.wallet.getAddress(),
+        // gasPrice: `0x${new Number(50 * 1000000000).toString(16)}`,
+        value: '0x' + toHex(toWei(Big(this.state.value), 'ether'))
+      };
+    }
+    const token = this.props.tokens.find(x => x.symbol === this.state.unit);
+    if (!token) {
+      return;
+    }
+
+    return {
+      to: token.address,
+      from: this.props.wallet.getAddress(),
+      // gasPrice: `0x${new Number(50 * 1000000000).toString(16)}`,
+      value: '0x0',
+      data: ERC20.transfer(
+        this.state.to,
+        Big(this.state.value).times(Big(10).pow(token.decimal))
+      )
+    };
+  }
+
+  estimateGas(state: State) {
+    this.props.node
+      .estimateGas(this.getTransactionFromState())
+      .then(gasLimit => {
+        if (this.state === state) {
+          let gasLimitString = gasLimit.toString();
+          if (gasLimitString === '21001' && state.unit === 'ether') {
+            gasLimitString = '21000';
+          }
+          if (gasLimit.gte(4000000)) {
```

Comment:
I can't find these conditions in v3, so I'm really confused about what's going on here. I'll move it out to a function somewhere in libs or utils, but it would be nice if @crptm could provide some insight into what's going on here.